### ISSUE TITLE
Named registrations in externally provided container are duplicated

### ISSUE THREAD

**Issue by SzymonPobiega** [Particular/NServiceBus.Unity#16]
### Symptoms

After providing a container with named registrations e.g.

```
container.RegisterType<IMyService, NamedService1>("1");
container.RegisterType<IMyService, NamedService2>("2");
container.RegisterType<IMyService, NamedService3>("3");
```

 to NServiceBus, calling `Builder.BuildAll<IMyService>()` will return 6 instead of 3 instances. Calling the container directly via `container.ResolveAll(typeof(IMyService))` will return 5 instances. All (but first when calling directly) registrations are duplicated.

### Who's affected

Affected are users who wish to do the following:
 * provide their own instance of Unity container 
 * use named registration feature to distinguish between various implementations of same service
 * at least in one place inject or resolve _all_ instances of the service

**Comment by SzymonPobiega** [comment]
@Particular/engineers please review

**Comment by ZlobnyiSerg** [comment]
I've also faced with this problem. Voting +1.

**Comment by SzymonPobiega** [comment]
@ZlobnyiSerg we've just released 6.1.1 https://github.com/Particular/NServiceBus.Unity/releases/tag/6.1.1
